### ISSUE TITLE
Otsu-Thresholding differs between CLIJ and clEsperanto

### ISSUE THREAD

**Issue by haesleinhuepf** [clEsperanto/pyclesperanto_prototype#43]
Dear all, dear future self,

when comparing CLIJ2 thresholding results, which uses core ImageJ under the hood, with results by pyclEsperanto using scikitimage, there are minor differences. The label-count differs by one because a single-pixel object is detected in python, while it's not in ImageJ.
I assume that's [related to an earlier discussion with @jni](https://forum.image.sc/t/otsus-threshold-method-delivers-different-results-in-scikit-image-and-imagej/24655/1) I'm documenting it here because I feel we should discuss options on how to fix this, and/or where to fix this. I assume we will hit more issue like that in the future and need a generic strategy ;-) 

![image](https://user-images.githubusercontent.com/12660498/99513231-abd4ae80-298a-11eb-84e7-1628261a66e1.png)

Reproduce the issue (CLIJ2 2.2.0.1, pyclesperanto-prototype 0.4.0):

ImageJ Macro:
```
// Init GPU
run("CLIJ2 Macro Extensions", "cl_device=");

// Load image from disc 
open("https://imagej.nih.gov/ij/images/blobs.gif");
image2 = getTitle();

// Gaussian Blur + Threshold Otsu
Ext.CLIJ2_push(image2);
Ext.CLIJ2_gaussianBlur2D(image2, image3, 1, 1);
Ext.CLIJ2_thresholdOtsu(image3, image4);
Ext.CLIJ2_connectedComponentsLabelingBox(image4, image5);

// show result
Ext.CLIJ2_pull(image5);

Ext.CLIJ2_getMaximumOfAllPixels(image5, number_of_labels);
print("Number of labels: " + number_of_labels);

Ext.CLIJ2_clear();
```

Python:
```
import pyclesperanto_prototype as cle
from skimage.io import imread

import numpy as np
import matplotlib
import matplotlib.pyplot as plt

# Load image
image = imread('https://imagej.nih.gov/ij/images/blobs.gif')

# Gaussian Blur + Threshold-Otsu
image1 = cle.push_zyx(image)
image2 = cle.subtract_image_from_scalar(image1, scalar=255)
image3 = cle.gaussian_blur(image2, sigma_x=1, sigma_y=1)
image4 = cle.threshold_otsu(image3)
image5 = cle.connected_components_labeling_box(image4)

# show result
plt.imshow(image5)
plt.show()

number_of_labels = cle.maximum_of_all_pixels(image5)
print("Number of labels: " + str(number_of_labels))
```

**Comment by cavenel** [comment]
Hi Robert,

Could that be because you don't add the minimum_intensity to the bin_centers in clEsperanto? It gives wrong result when minimum intensity of the image isn't zero.
https://github.com/clEsperanto/pyclesperanto_prototype/blob/master/pyclesperanto_prototype/_tier9/_threshold_otsu.py#L43
```python
bin_centers = np.arange(256) * range / (255)
```
vs (which I assume would be correct):
```python
bin_centers = np.arange(256) * range / (255) + minimum_intensity
```
![image](https://user-images.githubusercontent.com/6164758/151661047-4e243b61-0abe-4933-a36f-86682db69f56.png)

**Comment by haesleinhuepf** [comment]
Hi @cavenel ,

wow, that could absolutely make sense! Thanks for spotting this!

Have you tried it with the code provided above? If it works, I would be super happy if you send a pull-request with that change. 

Best,
Robert

**Comment by cavenel** [comment]
> Have you tried it with the code provided above?

Yes, that's what I tried in the small preview, and it gives 62 labels which seems similar to what CLIJ2 gives.
I will do a PR soon.

(The Otsu thresholding method giving me thresholds outside of the range of the input image was a good sign that something was not completely right :-p)

Christophe

**Comment by haesleinhuepf** [comment]
Fixed in #162
